### PR TITLE
JCLOUDS-1005: Retry on 500 and 503 errors

### PR DESCRIPTION
**Pull request by andrewgaul**
This requires rewriting the URL for `b2_upload_file` and `b2_upload_part`
requests.

### CODE REVIEW

**Comment by andrewgaul** [comment]
This reference explains retries:

https://www.backblaze.com/b2/docs/integration_checklist.html

A follow-on commit will reuse the authorization token as suggested.

File: b2/src/main/java/org/jclouds/b2/filters/B2RetryHandler.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+      return builder.build();
+   }
+
+   @Override
+   public boolean shouldRetryRequest(HttpCommand command, HttpResponse response) {
+      boolean retry = false;
+      try {
+         byte[] data = closeClientButKeepContentStream(response);
+         String message = data != null ? new String(data) : null;
```
Comment (nacx) [inline b2/src/main/java/org/jclouds/b2/filters/B2RetryHandler.java]: Message seems to be unused. Remove this line?

**Comment by nacx** [comment]
Just one small comment. LGTM!

File: b2/src/main/java/org/jclouds/b2/filters/B2RetryHandler.java (hunk 1 above)
Comment (andrewgaul) [inline b2/src/main/java/org/jclouds/b2/filters/B2RetryHandler.java]: Done.

**Pull request merged**